### PR TITLE
Resolved mode bug and added sleep clause for darwin systems. Fixes #7470

### LINKED ISSUE #7470: mode bug in tests (777 instead of 0o777)
testsuite.archiver:4531 (1.2-maint)

### PR DESCRIPTION
**Pull request by Michael-Girma**

### CODE REVIEW

File: src/borg/testsuite/archiver.py [hunk 1]
```diff
@@ -4523,12 +4523,16 @@ def test_time_diffs(self):
...
+            time.sleep(1)  # HFS has a 1s timestamp granularity
         self.create_regular_file("test_file", size=15)
         self.cmd('create', self.repository_location + '::archive2', 'input')
         output = self.cmd("diff", self.repository_location + "::archive1", "archive2")
         self.assert_in("mtime", output)
         self.assert_in("ctime", output)  # Should show up on windows as well since it is a new file.
-        os.chmod("input/test_file", 777)
+        if is_darwin:
+            time.sleep(1)  # HFS has a 1s timestamp granularity
+            os.chmod("input/test_file", 0o777)
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py:4535]: Must be outdented, tests are failing.

Comment (Michael-Girma) [reply src/borg/testsuite/archiver.py]: Sorry, I'm running darwin and the tests were passing on my local lol

**Review by ThomasWaldmann — APPROVED** [review]

**Comment by ThomasWaldmann** [comment]
Thanks!

**Comment by ThomasWaldmann** [comment]
Needs a fwd port to master also, I guess.

**Review by chandanpc — COMMENTED** [review]
it looks correct ,in chmod function one should use octal value. 777 is not an octal number.

**Pull request merged**